lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -119,28 +135,30 @@ def add_authorization_info(params, authorization)
         params[:method] = method
       end
 
+      def create_token(params, payment_method)
+        params[:apikey] = PARAMS_FOR_TOKENIZATION[:api_key]
+	      params[:js_security_key] = PARAMS_FOR_TOKENIZATION[:js_security_key]
+	      params[:ta_token] = PARAMS_FOR_TOKENIZATION[:ta_token]
+	      params[:callback] = PARAMS_FOR_TOKENIZATION[:callback]
+	      params[:type] = PARAMS_FOR_TOKENIZATION[:type]
+        params['credit_card.type'] = CREDIT_CARD_BRAND[payment_method.brand]
+        params['credit_card.cardholder_name'] = payment_method.name
+        params['credit_card.card_number'] = payment_method.number
+        params['credit_card.exp_date'] = format_exp_date(payment_method.month, payment_method.year)
+        params['credit_card.cvv'] = payment_method.verification_value if payment_method.verification_value?
```

Comment:
and/or could this be DRYed up with `use_creditcard`/`add_creditcard`?

Reply:
I looked at that but it would require a potentially more confusing refactor of `add_creditcard` due to other params being used & set in that method.

Reply:
I think it's reasonable for them to both call a separate method for the card data itself (something like `card_data`), adding a hash it returns to the appropriate element. I think some gateways do this, though I can't seem to find an example. It's not a big deal though.

Reply:
👍